From 396ff83f1fd2876785335f56a5007b8fb79c47a6 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Mon, 11 Sep 2023 15:22:23 +0300
Subject: [PATCH] [FSW-638] make: add falcon shores1 support

It is a multi architecure build (ARC and ARM)
ARC meant to be our main architecture, still need porting for arcboot
preboot and zephyr to have real content to be used by post scripts.
ARM is only planned to be used for CXL root port management.

Change-Id: Ib96fa5e8a57195535f5c26971305ef9b29d9adda
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/board/habanalabs/fs1/post-image-spi.sh b/board/habanalabs/fs1/post-image-spi.sh
new file mode 100755
index 0000000..fdc00d8
--- /dev/null
+++ b/board/habanalabs/fs1/post-image-spi.sh
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+# TODO fill with content
diff --git a/board/habanalabs/fs1_arc/post-image-spi.sh b/board/habanalabs/fs1_arc/post-image-spi.sh
new file mode 100755
index 0000000..d5fde5b
--- /dev/null
+++ b/board/habanalabs/fs1_arc/post-image-spi.sh
@@ -0,0 +1,93 @@
+#!/usr/bin/env bash
+
+NUM_PROD_KEYS=0
+
+function hl_copy_binaries {
+	# TODO fill with content
+
+	# creating temporary dummy files
+	touch fs1-erom.bin.le
+	touch fs1-erom_tpm.bin.le
+	touch fs1-ppboot-pcie_phy_fw.bin.le
+	touch fs1-preboot.bin.le
+	touch fs1-images-pointers.bin.le
+	touch fs1-fake-fuse.bin.le
+	touch preboot.bin
+	touch arcboot_rom.bin
+
+	return
+}
+
+function copy_all_keys {
+	# TODO fill with content
+	return
+}
+
+function generate_public_keys {
+	# TODO fill with content
+	return
+}
+
+function prep_image_pcie_pif {
+	# TODO fill with content
+	return
+}
+
+function prep_image_erom {
+	# TODO fill with content
+	return
+}
+
+function prep_image_fake_fuse {
+	# TODO fill with content
+	return
+}
+
+function zephyr_preboot_pack {
+	# TODO fill with content
+	return
+}
+
+function hl_join_bin {
+	# TODO fill with content
+	return
+}
+
+function hl_pack_bin {
+	# TODO fill with content
+	return
+}
+
+function prep_image_ptrs {
+	# TODO fill with content
+	return
+}
+
+function prep_image_spi_tpm {
+	# TODO fill with content
+	return
+}
+
+function calculate_hash {
+	# TODO fill with content
+
+	# creating temporary dummy hash result
+	touch hl_rsa3072.pub.hash
+
+	return
+}
+
+function create_spi_image {
+	# TODO fill with content
+
+	# creating temporary dummy image
+	touch fs1-spi-flash.bin.nocrc
+
+	return
+}
+
+function hl_pad_arcboot_rom {
+	# TODO fill with content
+
+	return
+}
diff --git a/configs/fs1_arc_defconfig b/configs/fs1_arc_defconfig
new file mode 100644
index 0000000..23ffa80
--- /dev/null
+++ b/configs/fs1_arc_defconfig
@@ -0,0 +1,15 @@
+BR2_USE_MMU=y
+BR2_arcle=y
+BR2_archs4x=y
+BR2_SVN="svn"
+BR2_DL_DIR="$(BR2_EXTERNAL_HABANALABS_PATH)/dl"
+BR2_SHARED_STATIC_LIBS=y
+BR2_PACKAGE_OVERRIDE_FILE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/local.mk"
+BR2_GLOBAL_PATCH_DIR="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/patches"
+BR2_TOOLCHAIN_EXTERNAL=y
+BR2_TARGET_GENERIC_HOSTNAME=""
+BR2_TARGET_GENERIC_ISSUE=""
+BR2_INIT_NONE=y
+BR2_SYSTEM_BIN_SH_NONE=y
+# BR2_TARGET_GENERIC_GETTY is not set
+BR2_ROOTFS_POST_IMAGE_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-spi-arc_global.sh"
diff --git a/configs/fs1_arm_defconfig b/configs/fs1_arm_defconfig
new file mode 100644
index 0000000..d5281dc
--- /dev/null
+++ b/configs/fs1_arm_defconfig
@@ -0,0 +1,16 @@
+BR2_aarch64=y
+BR2_SVN="svn"
+BR2_DL_DIR="$(BR2_EXTERNAL_HABANALABS_PATH)/dl"
+BR2_PACKAGE_OVERRIDE_FILE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/local.mk"
+# BR2_TOOLCHAIN is not set
+BR2_INIT_NONE=y
+BR2_SYSTEM_BIN_SH_NONE=y
+# BR2_TARGET_GENERIC_GETTY is not set
+# BR2_PACKAGE_BUSYBOX is not set
+# BR2_PACKAGE_IFUPDOWN_SCRIPTS is not set
+# BR2_TARGET_ROOTFS_TAR is not set
+BR2_PIC_PIE=n
+BR2_SSP_NONE=y
+BR2_RELRO_NONE=y
+BR2_FORTIFY_SOURCE_NONE=y
+BR2_TOOLCHAIN_USES_GLIBC=y
diff --git a/scripts/make/fs1/Makefile b/scripts/make/fs1/Makefile
new file mode 100644
index 0000000..2aa98ba
--- /dev/null
+++ b/scripts/make/fs1/Makefile
@@ -0,0 +1,2 @@
+.PHONY: fs1-post-image
+fs1-post-image:
diff --git a/scripts/make/hl-common/Makefile b/scripts/make/hl-common/Makefile
index e726528..abc65c8 100644
--- a/scripts/make/hl-common/Makefile
+++ b/scripts/make/hl-common/Makefile
@@ -1,5 +1,5 @@
-HL_MULTI_ARCH_RELEASE := gaudi2 gaudi3
-HL_DEFAULT_DEV_ARCH := gaudi2_arm gaudi3_arc
+HL_MULTI_ARCH_RELEASE := gaudi2 gaudi3 fs1
+HL_DEFAULT_DEV_ARCH := gaudi2_arm gaudi3_arc fs1_arc
 HL_SINGLE_ARCH_RELEASE :=
 HL_SECURITY_RELEASE :=
 
